src/x/headers/headers.go

```diff
@@ -76,6 +76,11 @@ const (
 	// metrics type.
 	MetricsStoragePolicyHeader = M3HeaderPrefix + "Storage-Policy"
 
+	// MetricsRestrictByPoliciesHeader provides the policies options to enforce
+	// on queries, in the form of a list of storage policies.
+	// "1m:14d;5m:60d"
+	MetricsRestrictByPoliciesHeader = M3HeaderPrefix + "Restrict-By-Policies"
```

Comment:
why `Policies` and not `Namespaces`?

Reply:
ah it's a storage policy? can we make the name more explicit to match `M3-Storage-Policy`

Reply:
Looks like we use storage policies as our general terminology everywhere, so went with policies.

Reply:
that's fine. but I think we should be more explicit and use `M3-Restrict-By-Storage-Policies`

Reply:
Ah yes, missed this. renaming [here](https://github.com/m3db/m3/pull/3719).

If both ways are set we fail the query.